rows.go

```diff
@@ -207,3 +208,14 @@ func (rows *snowflakeRows) waitForAsyncQueryStatus() error {
 	}
 	return nil
 }
+
+func (rows * snowflakeRows) addDownloader(newDL chunkDownloader) error {
```

Comment:
so other modules do not have to know the details about how downloaders are organized within rows.

Reply:
I don't think is particularly relevant, and the downloader logic is not available anyway since these functions remain unexported. I am dealing with chunk downloads at the moment so please revert them back.

Reply:
> I think there are a lot of changes here that are unnecessary and not related to this PR. Please keep the PR here minimal to the changes that are relevant.

Some changes are to better reuse code to avoid duplicating the same code everywhere.
addDownloader() can easily evolve to new logic/data structure without modifying other the callers. The two main processes in driver.go (construct conn object and use sc.cfg for authentication) are the same, the effects of these two functions are very straightforward and should be modularized - and needed when we want to use them in different places